### PR TITLE
Feature update memories form

### LINKED ISSUE #24: Edit a memory
As a user,
I should be able to click the edit svg on a memory then be prompted with a form that is already populated with the memory selected when I edit and save my changes a PATCH should be sent to the backend and the memory should re render with the updated information.

### LINKED ISSUE #27: Patch Request for Updating a Memory
As a user,
* I would like some functionality to change any of my logged memories.

### PR DESCRIPTION
**Pull request by cameronRomo**
## Description

* Adds a controlled form to update a memory
* Adds function definition for PATCH request on a memory
* Invokes said PATCH request for a successful patch

## Related Issue

#27 
#24 

## Motivation and Context

* There was no way for a user to update a memory, and it seemed like an important piece of functionality in regards to adding to or completely changing text related to a memory.

## How Has This Been Tested?

* Currently, we have logged the successful response and checked the postman to observe the desired changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/Memory/Memory.js [hunk 1]
```diff
@@ -9,23 +9,30 @@ import {
...
 
 export default Memory = ({ memory, setSelectedMemory }) => {
   const [modalVisible, setModalVisible] = useState(true);
+  const [editMode, setEditMode] = useState(false);
 
   return (
     <View style={styles.container}>
       <Text>CLICK ME</Text>
       <Modal presentationStyle='pageSheet' visible={modalVisible}>
+        <TouchableOpacity style={{height: 50, width: 50, position: "absolute", right: 100}} onPress={() => {
```
Comment (Codo-Baggins) [inline src/Memory/Memory.js:22]: We should put the inline style here into a styles object

File: src/Memory/Memory.js [hunk 2]
```diff
@@ -9,23 +9,30 @@ import {
...
+          console.log("Shit")
```
Comment (Codo-Baggins) [inline src/Memory/Memory.js:23]: We should remove this console.log....

File: src/MemoryForm/MemoryForm.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        style={styles.inputField}>
+        {memory.aromas}
+      </TextInput>
+      <Button onPress={()=> {
+        setUpdatedMemory({description: updatedDescriptionText, aromas: updatedAromaText})
+        // .then(() => console.log(updatedMemory))
+        // console.log(updatedMemory);
+        console.log("An API call happens here");
+        // editMemory(memory.id, updatedMemory)
+        // .then(response => console.log(response))
```
Comment (Codo-Baggins) [inline src/MemoryForm/MemoryForm.js:58]: We can remove all of this too

File: src/MemoryForm/MemoryForm.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    flexWrap: "wrap",
+    textAlign: "justify",
+    borderStyle: "solid",
+    borderColor: "#fc6d47",
+    height: 100,
+    width: "80%",
+    fontSize: 50,
+    borderWidth: 5,
+    flexShrink: 1,
+  },
```
Comment (Codo-Baggins) [inline src/MemoryForm/MemoryForm.js:87]: We should figure out which styles are exactly needed for this!

File: src/MemoryForm/MemoryForm.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  const [updatedAromaText, setUpdatedAromaText] = useState(memory.aromas);
+  const [updatedMemory, setUpdatedMemory] = useState({});
+  const [modalVisible, setModalVisible] = useState(true);
+
+  useEffect(() => {
+    if (Object.keys(updatedMemory).length) {
+      editMemory(memory.id, updatedMemory)
+      .then(response => console.log(response))
+    }
+  }, [updatedMemory])
```
Comment (Codo-Baggins) [inline src/MemoryForm/MemoryForm.js:32]: Good use of useEffect here to properly handle our PATCH!

**Review by Codo-Baggins — APPROVED** [review]
AWESOME job here.  Keep up the good work!

File: src/apiCalls.js [hunk 6]
```diff
@@ -13,4 +13,14 @@ export const getUser = async (id) => {
...
+}
+export const editMemory = async (id, memoryData) => {
+  const response = await fetch(`https://untangled-be.herokuapp.com/api/v1/memories/${id}`, {
+    method: "PATCH",
+    body: JSON.stringify({description: memoryData.description, aromas: memoryData.aromas}),
+    headers: {
+      "Content-type": "application/json; charset=UTF-8"
+    }
+  })
+  return await response.json();
```
Comment (brycemara) [inline src/apiCalls.js:25]: This is awesome. PATCH working great!

**Review by brycemara — APPROVED** [review]
This is awesome work team. Great PATCH and getting close to the MVP.

**Review by Estaffieri — APPROVED** [review]
The patch function is fantastic. I'll refine the styling for the team and get everything together! Thanks!

**Pull request merged**